packages/plugin-vue/src/index.ts

```diff
@@ -125,6 +125,9 @@ export default function vuePlugin(rawOptions: Options = {}): Plugin {
         },
         ssr: {
           external: ['vue', '@vue/server-renderer']
+        },
+        optimizeDeps: {
+          include: ['vue']
```

Comment:
This is something that VitePress currently does, but I think it may make more sense to actually exclude `vue` if we want to avoid reloads. As vue has a proper ESM entry for the browser to consume directly.

Maybe we should only add react, that is CJS and needs to be optimized for sure. And the vue one we can discuss it in another PR.

Reply:
👍 Done.